lib/pry/repl.rb

```diff
@@ -107,7 +107,7 @@ def read
         if output.tty? && pry.config.correct_indent && Pry::Helpers::BaseHelpers.use_ansi_codes?
           output.print @indent.correct_indentation(
             current_prompt, indented_val,
-            original_val.length - indented_val.length
+            [0, original_val.length - indented_val.length].max
```

Comment:
Not trying to be nitpicky here but this code is executed frequently, therefore it's better to make it more performant. I believe if we used a temp variable with an `if` check it would accomplish that goal. What do you think?

Reply:
Sorry for not being clear enough. I think we could put the result of evaluation of `original_val.length - indented_val.length` into a temp variable, then check if it’s larger than 0. If it is, then we return it. If not, then we return `nil`. With this approach there’s no need to allocate an array. Does it make sense?

Reply:
Oh I see, indeed a benchmark shows that an if is slightly faster, but allocating a temporary variable is actually slower:

~~~ rub
require 'benchmark'
array = (1..1000000).map { rand(10)-5 }
Benchmark.bmbm do |x|
  x.report("array") { array.each {|x| [x,0].max }}
  x.report("if")  { array.each {|x| x >= 0 ? x : 0  }}
  x.report("if temp var")  { array.each {|x| y=x; y=0 if y <= 0  }}             
end
~~~

                     user          system       total         real
array            0.048741   0.000000   0.048741  (0.048804)
if                   0.047802   0.000000   0.047802  (0.047877)
if temp var   0.052751   0.000000   0.052751  (0.052818)

Anyway I don't really think microoptimising this code is really important (especially since it follows an io operation), so I changed my version to your suggestion.

Reply:
Seems like you are right. Thanks for running this benchmark and sorry for the nitpicky comment.